File: grg/restore.js
const fs = require('fs');
const readline = require('readline');

const path = 'C:\\Users\\Dell\\.gemini\\antigravity\\brain\\c7fa05bd-004b-46bd-90ee-e1051a5bc837\\.system_generated\\logs\\transcript_full.jsonl';

const stream = fs.createReadStream(path, { encoding: 'utf8' });
const rl = readline.createInterface({ input: stream });

rl.on('line', (line) => {
  try {
    const entry = JSON.parse(line);
    if (entry.step_index === 31 && entry.type === 'GENERIC') {
      const content = entry.content || '';
      if (content.includes('File Path: ile:///c:/projetos/ai-engine-core/ai-engine/grg/public/runtime-cockpit.js')) {
        const lines = content.split('\n');
        const output = [];
        for (const l of lines) {
          const match = l.match(/^\d+: (.*)$/);
          if (match) output.push(match[1]);
        }
        fs.writeFileSync('public/runtime-cockpit.js', output.join('\n'));
        console.log('Restored runtime-cockpit.js: ' + output.length + ' lines');
      }
    }
    if (entry.step_index === 29 && entry.type === 'GENERIC') {
      const content = entry.content || '';
      if (content.includes('File Path: ile:///c:/projetos/ai-engine-core/ai-engine/grg/public/index.html')) {
        const lines = content.split('\n');
        const output = [];
        for (const l of lines) {
          const match = l.match(/^\d+: (.*)$/);
          if (match) output.push(match[1]);
        }
        fs.writeFileSync('public/index.html', output.join('\n'));
        console.log('Restored index.html: ' + output.length + ' lines');
      }
    }
    // Also unified.css at step 27?
    if (entry.step_index === 33 && entry.type === 'GENERIC') {
      const content = entry.content || '';
      if (content.includes('File Path: ile:///c:/projetos/ai-engine-core/ai-engine/grg/public/unified.css')) {
        const lines = content.split('\n');
        const output = [];
        for (const l of lines) {
          const match = l.match(/^\d+: (.*)$/);
          if (match) output.push(match[1]);
        }
        fs.writeFileSync('public/unified.css', output.join('\n'));
        console.log('Restored unified.css: ' + output.length + ' lines');
      }
    }
  } catch (e) {}
});

File: grg/restore.py
import json
import re

transcript_path = r'C:\Users\Dell\.gemini\antigravity\brain\c7fa05bd-004b-46bd-90ee-e1051a5bc837\.system_generated\logs\transcript_full.jsonl'

with open(transcript_path, 'r', encoding='utf-8') as f:
    for line in f:
        try:
            entry = json.loads(line)
            if entry.get('step_index') == 31 and entry.get('type') == 'GENERIC':
                content = entry.get('content', '')
                if 'File Path: ile:///c:/projetos/ai-engine-core/ai-engine/grg/public/runtime-cockpit.js' in content:
                    # extract lines starting with numbers
                    lines = content.split('\n')
                    output = []
                    for l in lines:
                        match = re.match(r'^\d+: (.*)$', l)
                        if match:
                            output.append(match.group(1))
                    
                    with open(r'c:\projetos\ai-engine-core\ai-engine\grg\public\runtime-cockpit.js', 'w', encoding='utf-8') as out:
                        out.write('\n'.join(output))
                    print('Restored runtime-cockpit.js ({} lines)'.format(len(output)))
                    
        except Exception as e:
            pass

with open(transcript_path, 'r', encoding='utf-8') as f:
    for line in f:
        try:
            entry = json.loads(line)
            if entry.get('step_index') == 29 and entry.get('type') == 'GENERIC':
                content = entry.get('content', '')
                if 'File Path: ile:///c:/projetos/ai-engine-core/ai-engine/grg/public/index.html' in content:
                    # extract lines starting with numbers
                    lines = content.split('\n')
                    output = []
                    for l in lines:
                        match = re.match(r'^\d+: (.*)$', l)
                        if match:
                            output.append(match.group(1))
                    
                    with open(r'c:\projetos\ai-engine-core\ai-engine\grg\public\index.html', 'w', encoding='utf-8') as out:
                        out.write('\n'.join(output))
                    print('Restored index.html ({} lines)'.format(len(output)))
                    
        except Exception as e:
            pass
